db-service/test/cqn4sql/where-exists.test.js

```diff
@@ -1567,6 +1567,107 @@ describe('path expression within infix filter following exists predicate', () =>
       `Only foreign keys of “genre” can be accessed in infix filter, but found “name”`
     )
   })
+  it('renders inner joins for the path expression at the leaf of scoped queries', () => {
+    let query = CQL`SELECT from bookshop.Authors:books[genre.name = null] { ID }`
+
+    const transformed = cqn4sql(query, model)
+    expect(transformed).to.deep.equal(
+      CQL`SELECT from bookshop.Books as books
+      inner join bookshop.Genres as genre on genre.ID = books.genre_ID
+      { books.ID }
+        WHERE EXISTS (
+          SELECT 1 from bookshop.Authors as Authors where Authors.ID = books.author_ID
+        )
+        and genre.name = NULL`,
+    )
+  })
+  it('renders nested inner joins for the path expression at the leaf of scoped queries', () => {
+    let query = CQL`SELECT from bookshop.Authors:books[genre.parent.name = null] { ID }`
```

Comment:
What about books that don't have a genre? Should they also appear in the result?

Reply:
hmm now that I think about it… it should have the same semantics as if you would write the expression into the queries `.where` clause. For this case, also books without a `genre` will be part of the result set due to the `left join`. What do you think?